### PR TITLE
Allow bloom filter for no dictionary SV columns

### PR DESCRIPTION
**Pull request by saurabhd336**
Instructions:
1. The PR has to be tagged with at least one of the following labels (*):
   1. `feature`
   2. `bugfix`
   3. `performance`
   4. `ui`
   5. `backward-incompat`
   6. `release-notes` (**)
2. Remove these instructions before publishing the PR.
 
(*) Other labels to consider:
- `testing`
- `dependencies`
- `docker`
- `kubernetes`
- `observability`
- `security`
- `code-style`
- `extension-point`
- `refactor`
- `cleanup`

(**) Use `release-notes` label for scenarios like:
- New configuration options
- Deprecation of configurations
- Signature changes to public methods/interfaces
- New plugins added or old plugins removed

### CODE REVIEW

File: pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/bloomfilter/BloomFilterHandler.java [hunk 1]
```diff
@@ -136,17 +138,132 @@ private void createBloomFilterForColumn(SegmentDirectory.Writer segmentWriter, C
...
         bloomFilterConfig);
-    try (BloomFilterCreator bloomFilterCreator = indexCreatorProvider.newBloomFilterCreator(
-        IndexCreationContext.builder().withIndexDir(indexDir).withColumnMetadata(columnMetadata)
-            .build().forBloomFilter(bloomFilterConfig));
-        Dictionary dictionary = getDictionaryReader(columnMetadata, segmentWriter)) {
-      int length = dictionary.length();
-      for (int i = 0; i < length; i++) {
-        bloomFilterCreator.add(dictionary.getStringValue(i));
+
+    if (columnMetadata.hasDictionary()) {
```
Comment (siddharthteotia) [inline pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/bloomfilter/BloomFilterHandler.java:142]: (nit) suggest separating this into different methods for readability. Something like ....

```
createBloomFilterOnDictColumn
createBloomFilterOnNoDictSVColumn
createBloomFilterOnNoDictMVColumn
```

File: pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/bloomfilter/BloomFilterHandler.java [hunk 2]
```diff
@@ -136,17 +138,132 @@ private void createBloomFilterForColumn(SegmentDirectory.Writer segmentWriter, C
...
+                bloomFilterCreator.add(
+                    BytesUtils.toHexString(forwardIndexReader.getBytes(i, readerContext))
+                );
+              }
+              break;
+            default:
+              throw new IllegalStateException("Unsupported data type: " + columnMetadata.getDataType() + " for column: "
+                  + columnMetadata.getColumnName());
+          }
+          bloomFilterCreator.seal();
```
Comment (siddharthteotia) [inline pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/bloomfilter/BloomFilterHandler.java:199]: can be called exactly once after the if branch for dict or no-dict

**Comment by siddharthteotia** [comment]
Suggest adding following tests -- Can be combined together where you generate some random data into a segment and then run queries on it or may be change any of the existing tests

- Segment generation with one or more raw SV columns with bloom filter
- Query execution tests on one or more raw SV columns with bloom filter.

**Comment by saurabhd336** [comment]
After further testing and discussions, came to realise that people generally move to no dictionary columns if they see high dictionary sizes. Introducing a hasSet to store unique values even for non dictionary columns, just to be able to support bloom filter (This PR was using the set just to provide the cardinality value when creating bloom filter) will be counterproductive. I'm closing this PR. Will rethink and raise a new one.

**Pull request closed without merging**